haystack/preview/components/rankers/top_p.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Union, Dict, Any
+
+import numpy as np
+from haystack.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(
+    message="Run 'pip install transformers[torch,sentencepiece]==4.32.1 sentence-transformers>=2.2.0'"
+) as torch_and_transformers_import:
+    import torch
+    from sentence_transformers import CrossEncoder
+    from haystack.modeling.utils import initialize_device_settings  # pylint: disable=ungrouped-imports
+
+
+@component
+class TopP:
```

Comment:
I was curious if a better name might be `TopPSampler` since often people often refer to sampling when talking about top p just as you do in the doc string. 

Also I'm a little unsure about calling this a ranker since it doesn't actually rerank anything I believe. From what I understand it is a good way of finding a dynamic cut-off as an alternative to using `top_k`. What do you think?

Reply:
Yeah, you are right on both; it'll be confusing to have it in rankers. I wanted to avoid creating a directory/package by moving it to rankers but I'll create samplers and home it there. Or perhaps is there another package name that could be more encompassing?

Reply:
I think samplers could be a good name and place for it for now. Maybe this will change as we add more types of these nodes and its clear what group they could be added to. For example, I could think of adding a similarity cutoff node that only returns docs above the threshold. This would be similar to top-p but instead rely on the raw sim scores instead of the of the softmax.